src/core/jsproxy.c

```diff
@@ -2650,11 +2650,9 @@ wrap_promise(JsVal promise, JsVal done_callback, PyObject* js2py_converter)
   result = _PyObject_CallMethodIdNoArgs(loop, &PyId_create_future);
   FAIL_IF_NULL(result);
 
-  helpers_mod = PyImport_ImportModule("_pyodide._future_helper");
-  FAIL_IF_NULL(helpers_mod);
```

Comment:
Looks like this import always fails in our test case. Maybe some proxy got early destroyed so that we fail to import here?

Reply:
It's not about a destroyed proxy. You can tell because if you change it to pass `__builtins__` it works:
```
    def g(x):
        import builtins
        return eval("x()", ChainMap({}, {"x": x, "__builtins__" : builtins}))
```

Reply:
I found the reason. In `PyImport_ImportModule`, it calls `__builtins__.__import__` internally.

When you are `await`ing a promis, you called `PyImport_ImportModule`, but the context don't have a `__builtins__`. This is reason.

Reply:
But why does `__builtins__` not get assigned to the globals? Whenever you eval with a dict as globals, Python will assign a `__builtins__` key if it's missing.

Reply:
Really interesting. In https://github.com/python/cpython/blob/3.12/Python/pythonrun.c#L1712-L1716 they makes a mistake similar as https://github.com/pyodide/pyodide/issues/4636#issuecomment-2017707331